src/vs/workbench/api/browser/mainThreadChatSessions.ts

```diff
@@ -370,8 +369,14 @@ export class MainThreadChatSessions extends Disposable implements MainThreadChat
 			provider,
 			onDidChangeItems: changeEmitter,
 		});
+
+		disposables.add(autorun(reader => {
+			const models = this._chatService.chatModels.read(reader);
+			disposables.add(this._chatSessionsService.registerModelProgressListener(Array.from(models), chatSessionType, changeEmitter));
```

Comment:
This will create and register a new listener each time the chat models change, which will leak memory. You may want to diff to get the added/removed listeners and just register those. Check out `autorunIterableDelta` to help with this, and then maybe have `DisposableMap` to add/remove the listener from each model individually